platform/linuxbsd/tts_linux.cpp

```diff
@@ -76,76 +81,84 @@ void TTS_Linux::speech_init_thread_func(void *p_userdata) {
 
 void TTS_Linux::speech_event_index_mark(size_t p_msg_id, size_t p_client_id, SPDNotificationType p_type, char *p_index_mark) {
 	TTS_Linux *tts = TTS_Linux::get_singleton();
-	if (tts && tts->ids.has(p_msg_id)) {
-		MutexLock thread_safe_method(tts->_thread_safe_);
-		// Get word offset from the index mark injected to the text stream.
-		String mark = String::utf8(p_index_mark);
-		DisplayServer::get_singleton()->tts_post_utterance_event(DisplayServer::TTS_UTTERANCE_BOUNDARY, tts->ids[p_msg_id], mark.to_int());
+	if (tts) {
+		tts->call_deferred(SNAME("_speech_index_mark"), p_msg_id, p_client_id, (int)p_type, String::utf8(p_index_mark));
```

Comment:
Would this work with `callable_mp(...).call_deferred()` instead of binding the methods?

Registering this with `GDCLASS` has made it show up in the GDExtension API: https://github.com/godotengine/godot-cpp/commit/c1ff169bf3ad5f13457eda7cd5a424b894adbb05#diff-fc336699721277f071b492b7c4006667e3073bb75cef6ddf75721ba8a61d12daR217376-R217382

Reply:
Yes, this will work as well - https://github.com/godotengine/godot/pull/73764